File: UniversityApi.API/Repositories/ProfessorsRepository.cs
using Microsoft.AspNetCore.Identity;
using Microsoft.EntityFrameworkCore;
using System.Net.Http;
using UniversityApi.API.Contracts;
using UniversityApi.API.DataBase;
using UniversityApi.API.DataBase.Entities;

namespace UniversityApi.API.Repositories
{
    public class ProfessorsRepository : GenericRepository<EntityProfessor>, IProfessorsRepository
    {
        public ProfessorsRepository(UniversityApiDbContext dbContext) : base(dbContext)
        {
        }

        public async Task<bool> IdCodeIsOccupied(string idCode)
        {
            return await _context.Set<EntityProfessor>().AnyAsync(s => s.IdCode.Equals(idCode));
        }
    }
}

File: FsUniversityApi/Repositories/ProfessorsRepository.fs
namespace FsUniversityApi.Repositories

open FsUniversityApi.Database.FsDbContext
open FsUniversityApi.Database.Entities.StudentCourseProfessor
open FsUniversityApi.Contracts
open Microsoft.EntityFrameworkCore;

type ProfessorsRepository(context: FsDbContext) =
    inherit GenericRepository<Professor>(context)

    interface IProfessorsRepository with
        member this.IdCodeIsOccupied (idCode : string) =
            context.Set<Professor>().AnyAsync(fun p -> p.IdCode.Equals(idCode));